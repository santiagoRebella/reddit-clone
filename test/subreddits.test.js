const React = require('react');
const immutable = require('immutable');
const Subreddits = require('components/subreddits');
const renderer = require('react-test-renderer');

jest.mock('react-router-dom', () => ({
  NavLink: 'NavLink'
}))

test('Subreddits is rendered correctly', () => {
  const component = renderer.create(
    <Subreddits
        list={immutable.List([{
          data: {
            id: "someid",
            title: "Test title",
            description: "Test Description",
            url: "Test Url",
            image: "Test image",
            display_name: "display_name"
          }}, {
          data: {
            id: "someotherid",
            title: "other Test title",
            description: "other Test Description",
            url: "other Test Url",
            image: "other Test image",
            display_name: "display_name2"
          }}, {
          data: {
            id: "anid",
            title: "A Test title",
            description: "A Test Description",
            url: "A Test Url",
            image: "A Test image",
            display_name: "display_name3"
          }}
        ])}
        subreddit="someSubreddit"
        onViewChange={jest.fn()}
        fetchSubreddits={jest.fn()}

    />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});

