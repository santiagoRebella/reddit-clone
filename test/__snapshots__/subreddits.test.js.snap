// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subreddits is rendered correctly 1`] = `
<section
  className="subreddits"
>
  <div
    className="card"
    style={
      Object {
        "display": "inline-block",
        "margin": 10,
      }
    }
  >
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        Test title
      </h5>
      <NavLink
        className="btn btn-primary"
        to="display_name"
      >
        display_name
      </NavLink>
    </div>
  </div>
  <div
    className="card"
    style={
      Object {
        "display": "inline-block",
        "margin": 10,
      }
    }
  >
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        other Test title
      </h5>
      <NavLink
        className="btn btn-primary"
        to="display_name2"
      >
        display_name2
      </NavLink>
    </div>
  </div>
  <div
    className="card"
    style={
      Object {
        "display": "inline-block",
        "margin": 10,
      }
    }
  >
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        A Test title
      </h5>
      <NavLink
        className="btn btn-primary"
        to="display_name3"
      >
        display_name3
      </NavLink>
    </div>
  </div>
</section>
`;
